trajopt_sco/src/solver_interface.cpp

```diff
@@ -132,21 +132,58 @@ std::ostream& operator<<(std::ostream& o, const Cnt& c)
 
 std::ostream& operator<<(std::ostream& o, const AffExpr& e)
 {
-  o << e.constant;
+  std::string sep = "";
+  if (e.constant != 0)
```

Comment:
Should [boost](https://www.boost.org/doc/libs/1_63_0/libs/math/doc/html/math_toolkit/float_comparison.html) be used for comparing floats?

Reply:
Everything else looks good.

Reply:
Hum I know float comparison is a tricky field, but in this case I think we should print exactly what is sent to the solver, without making assumptions as to what will be considered zero / non-zero. Using relative float comparison might for example hide issues with uninitialized values.

Reply:
Make sense in this case